src/pal/attestation.cpp

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+
+#if !defined(INSIDE_ENCLAVE) || defined(VIRTUAL_ENCLAVE)
+#  include "pal/attestation_snp.cpp"
+#  include "pal/attestation_virtual.cpp"
+#else
+#  include "pal/attestation_sgx.cpp"
+#endif
+
+namespace ccf::pal::attestation {
+
+#if !defined(INSIDE_ENCLAVE) || defined(VIRTUAL_ENCLAVE)
```

Comment:
Would it be possible to reduce the number of times we have this compile-time pattern in the code base and use this instead, e.g.:
```c++
#if !defined(INSIDE_ENCLAVE) || defined(VIRTUAL_ENCLAVE)
  using AttestationImpl = SnpAttestation;
#else
  using AttestationImpl = SgxAttestation;
#endif

ReportData make_report_data(crypto::Sha256Hash& node_public_key_hash) 
{
    AttestationImpl::make_report_data(node_public_key_hash);
}
```